src/mono/mono/metadata/CMakeLists.txt

```diff
@@ -5,7 +5,7 @@ project(metadata C)
 #
 if(DISABLE_ICALL_TABLES)
 add_library(mono-icall-table STATIC "icall-table.c")
-target_link_libraries(mono-icall-table PRIVATE monoapi eglib_api)
+target_link_libraries(mono-icall-table PRIVATE monoapi eglib_api zlibstatic)
```

Comment:
Is there a better way to import zlibstatic here? Maybe link it once for everyone instead of including it on each `target_link_libraries` call?

Reply:
This is the correct way to do this.

Reply:
If you do `target_link_libraries(libname PUBLIC zlibstatic)` then you can remove some of the cases in the other mono files (as they will flow transitively)

Reply:
I didn't put this comment in the right place. I actually meant all the changes at the end of this file. But if the current way is the acceptable way of doing it, and since it works, I don't see why I should change it.